pkg/solana/ccip/chainaccessor/cache.go

```diff
@@ -0,0 +1,85 @@
+package chainaccessor
+
+import (
+	"sync"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type pdaCache struct {
+	offrampCache     offrampPDACache
+	offrampCacheMu   sync.RWMutex
+	feeQuoterCache   feeQuoterPDACache
+	feeQuoterCacheMu sync.RWMutex
+	routerCache      routerPDACache
+	routerCacheMu    sync.RWMutex
+	rmnRemoteCache   rmnRemotePDACache
+	rmnRemoteCacheMu sync.RWMutex
+}
+
+type offrampPDACache struct {
+	state        solana.PublicKey
+	config       solana.PublicKey
+	refAddresses solana.PublicKey
+	sourceChain  map[uint64]solana.PublicKey
+}
+
+type feeQuoterPDACache struct {
+	config             solana.PublicKey
+	billingTokenConfig map[solana.PublicKey]solana.PublicKey
+	destChainConfig    map[uint64]solana.PublicKey
+}
+
+type routerPDACache struct {
+	config         solana.PublicKey
+	destChainState map[uint64]solana.PublicKey
+	// TODO: we should maybe calculate this on the fly to avoid having to cache an address for all users for every chain selector
+	nonce map[uint64]map[solana.PublicKey]solana.PublicKey
+}
+
+type rmnRemotePDACache struct {
+	curse solana.PublicKey
+}
+
+func newPDACache() pdaCache {
+	return pdaCache{
+		offrampCache:     newOfframpPDACache(),
+		offrampCacheMu:   sync.RWMutex{},
```

Comment:
Should be enough to just group all of these under the same mutex, since updates will be rare once the initial sync concludes